### PR TITLE
MTDSA-6856 Validation and Additional Tests

### PR DESCRIPTION
**Pull request by Nboaram**

### CODE REVIEW

File: app/utils/CurrentDateTime.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package utils
+
+import javax.inject.{Inject, Singleton}
+import org.joda.time.{DateTime, DateTimeZone}
+
+@Singleton
+class CurrentDateTime @Inject()() {
```
Comment (Mbrookesy) [inline app/utils/CurrentDateTime.scala:23]: Seems to be no coverage for this

File: app/utils/CurrentTaxYear.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+package utils
+
+import javax.inject.Singleton
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+
+@Singleton
+class CurrentTaxYear {
```
Comment (Mbrookesy) [inline app/utils/CurrentTaxYear.scala:24]: Ditto for no coverage

File: app/v1/controllers/requestParsers/validators/AmendOtherDeductionsValidator.scala [hunk 3]
```diff
@@ -16,13 +16,17 @@
...
 
+import config.AppConfig
+import javax.inject.Inject
+import utils.CurrentTaxYear
 import v1.controllers.requestParsers.validators.validations._
 import v1.models.errors.{MtdError, RuleIncorrectOrEmptyBodyError}
 import v1.models.request.amendOtherDeductions.{AmendOtherDeductionsBody, AmendOtherDeductionsRawData, Seafarers}
 
-class AmendOtherDeductionsValidator extends Validator[AmendOtherDeductionsRawData] {
+class AmendOtherDeductionsValidator @Inject()(implicit appConfig: AppConfig, currentTaxYear: CurrentTaxYear)
```
Comment (Mbrookesy) [inline app/v1/controllers/requestParsers/validators/AmendOtherDeductionsValidator.scala:26]: Coverage seems low for the validator

File: app/v1/controllers/requestParsers/validators/validations/package.scala [hunk 4]
```diff
@@ -22,5 +22,6 @@ package object validations {
 
   val dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd")
   val NoValidationErrors = List()
+  val minimumTaxYear = 2019
```
Comment (Mbrookesy) [inline app/v1/controllers/requestParsers/validators/validations/package.scala:25]: There doesn't seem to be any use for this variable

File: test/v1/controllers/requestParsers/validators/validations/TaxYearValidationSpec.scala [hunk 5]
```diff
@@ -17,7 +17,7 @@
 package v1.controllers.requestParsers.validators.validations
 
 import support.UnitSpec
-import v1.models.errors.{RuleTaxYearRangeInvalidError, TaxYearFormatError}
+import v1.models.errors.{RuleTaxYearNotSupportedError, RuleTaxYearRangeInvalidError, TaxYearFormatError}
 import v1.models.utils.JsonErrorValidators
 
 class TaxYearValidationSpec extends UnitSpec with JsonErrorValidators {
```
Comment (Mbrookesy) [inline test/v1/controllers/requestParsers/validators/validations/TaxYearValidationSpec.scala:23]: Seems to be an unusued import here

**Review by Mbrookesy — CHANGES_REQUESTED** [review]

File: app/v1/controllers/requestParsers/validators/AmendOtherDeductionsValidator.scala (hunk 3 above)
Comment (Nboaram) [reply app/v1/controllers/requestParsers/validators/AmendOtherDeductionsValidator.scala:26]: The coverage is low with the body format validation, I didn't touch that. And I'm not sure how I would raise it.

**Pull request merged**